concourse/steps/update_component_deps.py

```diff
@@ -205,18 +205,44 @@ def determine_reference_versions(
         raise NotImplementedError
 
 
+def create_ctx_repo_mapping(
+    map: dict[str, str],
+) -> dict[str, gci.componentmodel.OciRepositoryContext]:
+    # invert the input map and replace the ctx repo names with actual instances, i.e. turn it from
+    # a mapping of ctx repo name -> component name(s) to a mapping of component name -> ctx repo.
+    cfg_factory = ci.util.ctx().cfg_factory()
+
+    return {
+        component_name: gci.componentmodel.OciRepositoryContext(
+            baseUrl=cfg_factory.ctx_repository(ctx_repo_name).base_url()
+        )
+        for ctx_repo_name, component_name_list in map.items()
+            for component_name in component_name_list
+    }
+
+
 def determine_upgrade_prs(
     upstream_component_name: str,
     upstream_update_policy: UpstreamUpdatePolicy,
     upgrade_pull_requests: typing.Iterable[github.util.UpgradePullRequest],
-    ctx_repo: gci.componentmodel.OciRepositoryContext,
+    default_ctx_repo: gci.componentmodel.OciRepositoryContext,
```

Comment:
I would prefer to remove `default_ctx_repo` and always pass-in a resolver-callable; ocm-repo calculation should rather be done by caller (dependency-injection)